cmd/prometheus/main.go

```diff
@@ -627,6 +634,21 @@ func main() {
 		discoveryManagerNotify = legacymanager.NewManager(ctxNotify, log.With(logger, "component", "discovery manager notify"), legacymanager.Name("notify"))
 	}
 
+	var secretManager *secrets.Manager
+	if cfg.enableKubeSecretProvider {
+		manager := secrets.NewManager(
+			ctxSecrets,
+			prometheus.DefaultRegisterer,
+			secrets.ProviderOptions{
+				Logger: log.With(logger, "component", "secret manager"),
+			},
+		)
+		secretManager = &manager
+		defer secretManager.Close(prometheus.DefaultRegisterer)
+	}
+
+	cfg.scrape.HTTPClientOptions = append(cfg.scrape.HTTPClientOptions, common_config.WithSecretManager(secretManager))
```

Comment:
question: In the case where the flag is disabled, I'm assuming it's safe to add a `nil` reference for the uninitialized `secretManager` pointer?

What about just not adding the manager at all in that case?

Reply:
Good catch. Thanks!